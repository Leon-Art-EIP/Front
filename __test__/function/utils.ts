import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
