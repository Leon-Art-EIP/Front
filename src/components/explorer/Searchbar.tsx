import ClearIcon from "@mui/icons-material/Clear";
import SearchIcon from "@mui/icons-material/Search";
import { useState } from "react";

interface SearchbarProps {
  handleSearchTerm: (searchTerm: string) => void;
}

export default function Searchbar(props: SearchbarProps): JSX.Element {
  const [searchTerm, setSearchTerm] = useState<string>("");

  function onSearchTerm(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
    props.handleSearchTerm(searchTerm);
  }

  return (
    <form className="relative w-full" onSubmit={onSearchTerm}>
      <input
        className="w-full h-14 bg-secondary rounded-3xl pl-16 focus:shadow-md focus:outline-none focus:ring-1 focus:ring-tertiary transition duration-300 ease-in-out text-tertiary"
        placeholder="Rechercher une oeuvre ou un utilisateur..."
        onChange={(e) => setSearchTerm(e.target.value)}
        value={searchTerm}
      />
      <button
        type="submit"
        className="p-2 absolute top-2 left-4 hover:opacity-75 rounded-xl transition hover:bg-secondary hover:shadow-lg duration-300 ease-in-out"
      >
        <SearchIcon className={`text-tertiary`} />
      </button>
      {searchTerm && (
        <button className="absolute top-2 right-2">
          <ClearIcon className="text-primary" />
        </button>
      )}
    </form>
  );
}
