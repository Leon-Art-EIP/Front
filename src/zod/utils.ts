export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
