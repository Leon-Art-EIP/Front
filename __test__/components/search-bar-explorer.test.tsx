import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Searchbar from "../../src/components/explorer/Searchbar";

test("Search bar explorer snapshot", () => {
  testSnapshot(<Searchbar handleSearchTerm={() => {}} />);
});
