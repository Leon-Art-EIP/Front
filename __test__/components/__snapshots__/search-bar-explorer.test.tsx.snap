// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search bar explorer snapshot 1`] = `
<form
  className="relative w-full"
  onSubmit={[Function]}
>
  <input
    className="w-full h-14 bg-secondary rounded-3xl pl-16 focus:shadow-md focus:outline-none focus:ring-1 focus:ring-tertiary transition duration-300 ease-in-out text-tertiary"
    onChange={[Function]}
    placeholder="Rechercher une oeuvre ou un utilisateur..."
    value=""
  />
  <button
    className="p-2 absolute top-2 left-4 hover:opacity-75 rounded-xl transition hover:bg-secondary hover:shadow-lg duration-300 ease-in-out"
    type="submit"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-tertiary css-i4bv87-MuiSvgIcon-root"
      data-testid="SearchIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </button>
</form>
`;
